### PR TITLE
Support Elastic License 2.0

### PR DESCRIPTION
**Pull request by marclop**
## Description

Adds support for the Elastic License 2.0 as `Elasticv2`, also updating
the tests to support multiple license rewrite tests.

## Related issues

Part of elastic/apm-server#6105

### CODE REVIEW

**Comment by ruflin** [comment]
Great to see this happening! We will need this also in other places.

File: main_test.go [hunk 1]
```diff
@@ -93,6 +93,38 @@ testdata/singlelevel/main.go: is missing the license header
...
+testdata/multilevel/sublevel/autogen.go: is missing the license header
+testdata/multilevel/sublevel/doc.go: is missing the license header
+testdata/multilevel/sublevel/partial.go: is missing the license header
+testdata/singlelevel/doc.go: is missing the license header
+testdata/singlelevel/main.go: is missing the license header
+testdata/singlelevel/wrapper.go: is missing the license header
+testdata/singlelevel/zrapper.go: is missing the license header
+testdata/x-pack/doc.go: is missing the license header
+testdata/x-pack/wrong.go: is missing the license header
+testdata/x-pack-v2/doc.go: is missing the license header
```
Comment (axw) [inline main_test.go:126]: Is it a mistake that testdata/x-pack-v2/doc.testdata does not have the ELv2 license header? Why do we have "correct.testdata", "doc.testdata", and "wrong.testdata" in x-pack-v2, but only "doc.testdata" and "wrong.testdata" in x-pack?

**Review by axw — APPROVED** [review]
LGTM apart from some surprising testdata

File: main_test.go (hunk 1 above)
Comment (marclop) [reply main_test.go:126]: actually, I meant to remove doc and leave only `correct` and `wrong`. I'll update the PR.

**Pull request merged**